### ISSUE TITLE
Checkbox Groups have different validation requirements for JS vs PHP

### ISSUE THREAD

**Issue by iainhenderson** [verbb/formie#553]
**Description**
Formie will allow a required checkbox group to submit if a single checkbox is checked, but the JS validation messaging doesn't clear until all checkboxes are checked.

Additionally multiple times I have managed to have the checkbox group show multiple JS errors while selecting deselecting options. But I have not come up with a solid way to reproduce that issue.

**Steps to reproduce**
1. Create a form with required checkboxes and 'Validate When Typing'
2. Submit the form - the JS error message displays.
3. Select a checkbox option, you can now submit the form but the messaging hasn't been cleared.
4. Select all checkbox options and see the JS error dissapear.

### Form settings
- Multi-page form: Yes
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: Yes

**Additional info**
- Plugin version: 1.4.13
- Craft version: 3.7.8
- Multi-site: No

**Additional context**

Here is a video demonstrating the issue:

https://user-images.githubusercontent.com/775758/130725043-ec1c003f-a3f6-42a7-9b78-1d771df822bd.mov

Here is a screenshot showing the custom settings. We're replacing the CSS, but using the validation JS.

![Screen Shot 2021-08-25 at 1 46 32 pm](https://user-images.githubusercontent.com/775758/130725157-0afa6438-f361-4efe-af6f-cd40d8380ca4.png)

Here is a screenshot showing the multiple error issue:
![Screen Shot 2021-08-25 at 1 41 35 pm](https://user-images.githubusercontent.com/775758/130725407-88dac8a2-ac58-4e11-81af-7660fa79b07b.png)

**Comment by engram-design** [comment]
Quick question - you note the use of custom templates, but in your form templates, you're not using overrides. Can you confirm if you're doing anything custom with the checkboxes output? The issue of validation messages showing multiple times has cropped up a few times in the past, but usually related to custom templates.

Having said that, looks like there's an issue with the "Validate When Typing" setting and this behaviour. Fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.14",
  "...": "..."
}
```

Then run `composer update`.

**Comment by iainhenderson** [comment]
The custom template just loads different CSS, the HTML and JS is unmodified. To test, I changed the form back to use "Default Formie Template" and the same behaviour occurs.

Using the "dev-craft-3 as 1.4.14" version fixes the issue. I've also tried replicating the multiple error issue (shown at the end of my original comment) and can no longer seem to trigger it, so I think that fixed too.

Thanks heaps for sorting that out @engram-design
